pytest_mergify/tests_fetcher.py

```diff
@@ -0,0 +1,52 @@
+import dataclasses
+import typing
+
+import requests
+
+from pytest_mergify import utils
+
+
+@dataclasses.dataclass
+class TestsFetcher:
+    api_url: str
+    token: str
+    repo_name: str
+    branch_name: str
+    test_names: typing.List[str] = dataclasses.field(init=False, default_factory=list)
+    init_error_message: typing.Optional[str] = dataclasses.field(
+        init=False, default=None
+    )
+
+    def __post_init__(self) -> None:
+        try:
+            owner, repository = utils.split_full_repo_name(self.repo_name)
+        except utils.InvalidRepositoryFullNameError:
+            self.init_error_message = f"Repository name '{self.repo_name}' has an unexpected format (expected 'owner/repository')"
+            return
+
+        try:
+            response: requests.Response = requests.get(
+                url=f"{self.api_url}/v1/ci/{owner}/tests/names",
+                headers={"Authorization": f"Bearer {self.token}"},
+                params={"repository": repository, "branch": self.branch_name},
+                timeout=10,
+            )
+        except requests.ConnectionError as exc:
+            self.init_error_message = (
+                f"Failed to connect to Mergify's API. Error: {str(exc)}"
+            )
+            return
+
+        if response.status_code in {402, 404}:
+            # Feature not available or endpoint missing; treat as silent skip.
+            return
```

Comment:
All users using CI insights should have access to this API, right?  Maybe it's safer to let the code below handle the error or it will be hard to debug in case of issue